addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/handler/NestBaseHandler.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.nest.handler;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Optional;
+import java.util.TimeZone;
+
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.nest.internal.NestDeviceDataListener;
+import org.openhab.binding.nest.internal.NestIdentifiable;
+import org.openhab.binding.nest.internal.NestUpdateRequest;
+import org.openhab.binding.nest.internal.config.NestDeviceConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Deals with the structures on the Nest API, turning them into a thing in openHAB.
+ *
+ * @author David Bennett - initial contribution
+ * @author Martin van Wingerden - Splitted of NestBaseHandler
+ */
+abstract class NestBaseHandler extends BaseThingHandler implements NestDeviceDataListener, NestIdentifiable {
+    private final Logger logger = LoggerFactory.getLogger(NestBaseHandler.class);
+
+    NestBaseHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing handler for {}", getClass().getName());
+        getNestBridgeHandler().map(b -> b.addDeviceDataListener(NestBaseHandler.this));
+
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "Waiting for refresh");
+    }
+
+    @Override
+    public void dispose() {
+        getNestBridgeHandler().map(b -> b.removeDeviceDataListener(NestBaseHandler.this));
+    }
+
+    @Override
+    public String getId() {
+        return getDeviceId();
+    }
+
+    private String getDeviceId() {
+        return getConfigAs(NestDeviceConfiguration.class).deviceId;
+    }
+
+    private Optional<NestBridgeHandler> getNestBridgeHandler() {
+        return Optional.ofNullable(getBridge()).map(b -> (NestBridgeHandler) b.getHandler());
```

Comment:
I honestly think that these constructs are rather ugly for rather simply code.
Note that in the discussion around coding guidelines, we opted against using Optionals and instead use null annotations in the code (where appropriate). Optionals should be limited to APIs where the use of streams makes sense, but this is obviously not the case here.

Reply:
Rewritten to traditional null checks.